pyproject.toml

```diff
@@ -27,7 +27,7 @@ classifiers = [
 
 [tool.poetry.dependencies]
 async-timeout = { version = ">=3.0.0", python = "<3.11" }
-typing-extensions = { version = ">=4.7.0", python = "<3.12" }
+typing-extensions = { version = ">=4.7.0" }
```

Comment:
Ideally, we would drop the `typing-extensions` dependency in a few years when we drop support for Python 3.11. This is why we have the extra conditionals in the code and here.